### DISCUSSION TITLE
ONEDRIVE_MONITOR docker environment variable

### DISCUSSION THREAD

**Discussion by eelcofolkertsma** [abraunegg/onedrive#2795] [Q&A]
Docker image is hard-coded to --monitor in /entrypoint.sh. It feels straightforward to introduce an environment variable to switch between --monitor (true) and --synchronize (false). 
But the client is very rich and well thought out, and I am very much a newbie. Is there a simple reason to not have such a switch in the script?

**Comment by abraunegg** [comment]
Simple reason is your Docker container would then simply exit after its first sync and that is not the point of running the Docker container.

**Reply by eelcofolkertsma** [reply]
Let me share my use case. 
My home server is a raspberry pi. Every 18 months or so I need to move to next generation of operating system. The recommended process is to start from scratch, wipe system drive and load fresh image of OS. So every cycle I need to reinstall all my applications and that is significant work. My solution: leverage docker to avoid app-installs to the max. And that applies both to "server" use and to tools I use for "ad-hoc" like an editor or a file transfer tool.
My cloud storage is OneDrive (as I need Office for my day job). It hosts a large collection of music and photo's. For hobby I sometimes need some photo's, some music on the raspberry. That is where "onedrive --synchronize" helps me very, very well (Thank you!!!)

Here is example of community where Docker Image is perceived as full alternative to local install https://www.asyncapi.com/docs/tools/cli/installation 

I shall go ahead and make small modification to entrypoint.sh so I can launch the image for --synchronize as well. Your answer gives me faith this might work
